### PR TITLE
Allow program to work without flag

### PR DESCRIPTION
**Pull request by enzanki-ars**
If env var $URL is set, there is no need for the -url.
-url will override $URL if set.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -55,6 +55,9 @@ func init() {
 
 	// Parse all flags
 	flag.Parse()
+	if slackURL == "" {
```
Comment (jackzampolin) [inline main.go:59]: Can you make it quit with an error if there is no slack url passed through the flag or through a ENV variable?

maybe:

``` go
if slackURL == "" {
  slackURL = os.Getenv("CATSLACK_URL")
  if slackURL == "" {
    fmt.Println("Need to pass in slack url via the -url flag or the $SLACK_URL env variable")
    os.Exit(1)
  }
}
```

**Comment by jackzampolin** [comment]
Once inline comment is addressed I will merge

**Comment by enzanki-ars** [comment]
~~I added some more commits, and fixed the eventual merge conflict, but GitHub seems to be [having issues](https://status.github.com/messages) displaying them right now.~~ (Fixed)

File: main.go [hunk 2]
```diff
@@ -55,14 +55,19 @@ func init() {
...
 
 func main() {
 
 	// Exit if not passed a url
 	if slackURL == "" {
-		fmt.Println("Must pass a working Slack integration URL using the `-url` flag.")
-		return
+		fmt.Println("Must pass a working Slack integration URL using the `-url` flag")
+		fmt.Println("or by setting the $CATSLACK_URL env variable.")
+		os.Exit(1)
```
Comment (enzanki-ars) [inline main.go:70]: Already has this exit code though... The `return` statement already had this, though I did change it to os.Exit to stop the program with a controlled exit code.

**Pull request merged**